src/binder_util.c

```diff
@@ -815,6 +853,47 @@ binder_strv_from_hidl_string_vec(
     return NULL;
 }
 
+gboolean
+binder_read_string16_parse_int(
+    GBinderReader* reader,
+    gint32* value)
+{
+    /* Read a string and parse integer value from it */
+    gboolean ret = FALSE;
+    char* str = gbinder_reader_read_string16(reader);
+
+    if (str) {
+        ret = gutil_parse_int(str, 10, value);
+        g_free(str);
```

Comment:
Would work even when `NULL` but matter of taste...

Reply:
Removed the str NULL check.